activerecord/lib/active_record/errors.rb

```diff
@@ -353,20 +353,24 @@ class Deadlocked < TransactionRollbackError
   class IrreversibleOrderError < ActiveRecordError
   end
 
+  # Superclass for all timeout-related errors
+  class Timeout < StatementInvalid
```

Comment:
Does it make sense that Timeout still inherits from `StatementInvalid`? It was previously like this for backward compatibility reasons , but does it still apply now ? cc/ @byroot

Reply:
Yes. We still need to keep it backward compatible.